qdrant-landing/content/documentation/tutorials/neural-search.md

```diff
@@ -0,0 +1,338 @@
+
+</aside>
+
+4. Create an iterator over the startup data and vectors.
+
+The Qdrant client library defines a special function that allows you to load datasets into the service.
+However, since there may be too much data to fit a single computer memory, the function takes an iterator over the data as input.
+
+```python
+import numpy as np
+import json
+
+fd = open('./startups.json')
+
+# payload is now an iterator over startup data
+payload = map(json.loads, fd)
+
+# Load all vectors into memory, numpy array works as iterable for itself.
+# Other option would be to use Mmap, if you don't want to load all data into RAM
+vectors = np.load('./startup_vectors.npy')
+```
+
+5. Upload the data
+
+```python
+qdrant_client.upload_collection(
+    collection_name='startups',
+    vectors=vectors,
+    payload=payload,
+    ids=None,  # Vector ids will be assigned automatically
+    batch_size=256  # How many vectors will be uploaded in a single request?
+)
+```
+
+Vectors are now uploaded to Qdrant. 
+
+## Build the search API
+
+Now that all the preparations are complete, let's start building a neural search class.
+
+First, install all the requirements:
+
+```bash
+pip install sentence-transformers numpy
+```
+
+In order to process incoming requests, neural search will need 2 things: 1) a model to convert the query into a vector and 2) the Qdrant client to perform search queries.
+
+1. Create a file named `neural_searcher.py` and specify the following.
+
+```python
+
+from qdrant_client import QdrantClient
+from sentence_transformers import SentenceTransformer
+
+
+class NeuralSearcher:
+
+    def __init__(self, collection_name):
+        self.collection_name = collection_name
+        # Initialize encoder model
+        self.model = SentenceTransformer('all-MiniLM-L6-v2', device='cpu')
+        # initialize Qdrant client
+        self.qdrant_client = QdrantClient('http://localhost:6333')
+```
+
+2. Write the search function.
+
+```python
+    def search(self, text: str):
+        # Convert text query into vector
+        vector = self.model.encode(text).tolist()
+
+        # Use `vector` for search for closest vectors in the collection
+        search_result = self.qdrant_client.search(
+            collection_name=self.collection_name,
+            query_vector=vector,
+            query_filter=None,  # If you don't want any filters for now
+            top=5  # 5 the most closest results is enough
+        )
+        # `search_result` contains found vector ids with similarity scores along with the stored payload
+        # In this function you are interested in payload only
+        payloads = [hit.payload for hit in search_result]
+        return payloads
+```
+
+3. Add search filters.
+
+With Qdrant it is also feasible to add some conditions to the search.
+For example, if you wanted to search for startups in a certain city, the search query could look like this:
+
+```python
+from qdrant_client.models import Filter
```

Comment:
This comes from the `http` submodule
```suggestion
from qdrant_client.http.models import Filter
```

Reply:
nope, `from qdrant_client.models import Filter` should work

Reply:
http model will also work, but it is unnecessary specifics